File: src/main/java/io/tripled/social/client/presentation/controller/PostController.java
package io.tripled.social.client.presentation.controller;

import io.tripled.social.client.application.PostMessageUseCase;
import io.tripled.social.client.presentation.Output;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class PostController extends AbstractController {

  private PostMessageUseCase postMessageUseCase;

  public PostController(PostMessageUseCase postMessageUseCase) {
    this.postMessageUseCase = postMessageUseCase;
  }

  @Override
  protected Pattern getPattern() {
    return Pattern.compile("(.+) -> (.+)");
  }

  @Override
  public void execute(String inputLine, Output output) {
    Matcher matcher = getPattern()
        .matcher(inputLine.trim());

    if (matcher.matches()) {
      String userName = matcher.group(1).trim();
      String message = matcher.group(2).trim();

      postMessageUseCase.postMessage(userName, message);
    }
  }
}

File: src/main/kotlin/io/tripled/social/client/presentation/controller/PostController.kt
package io.tripled.social.client.presentation.controller

import io.tripled.social.client.application.PostMessageUseCase
import io.tripled.social.client.presentation.Controller
import io.tripled.social.client.presentation.Output
import io.tripled.social.client.presentation.request.PostCliRequest

class PostController(private val postMessageUseCase: PostMessageUseCase) : Controller<PostCliRequest> {
    override fun execute(request: PostCliRequest, output: Output) {
        output.print(postMessageUseCase.postMessage(request.userName, request.message))
    }
}
